users.js

```diff
@@ -640,6 +677,27 @@ class User {
 		}
 		return true;
 	}
+	lockName() {
+		let userid = this.userid;
+		for (let ip in this.ips) {
+			nameLockedIps[ip] = userid;
+		}
+		if (this.autoconfirmed) nameLockedUsers[this.autoconfirmed] = userid;
+		nameLockedUsers[this.userid] = userid;
+		this.namelocked = userid;
+		this.forceRename(this.genericName(), this.registered);
+		this.updateIdentity();
+
+		return true;
+	}
+	genericName() {
+		let num = '' + this.guestNum;
+		// Make sure the name is too long for regular name choosing.
+		while (num.length < 6) {
+			num = '0' + num;
+		}
+		return 'Pokemon Trainer #' + num;
+	}
```

Comment:
I'm not sure I like the idea of this being static just because it doesn't use `this`. If it's intended only for use by User instances, having it be an instance method seems fine to me.

If it's not intended for use inside a single User instance, having it non-static might be better? If it's static, it at least shouldn't be in the middle of a bunch of non-static methods.